### ISSUE TITLE
[BUG] PPO example made no training progress

### ISSUE THREAD

**Issue by SiliconSloth** [pytorch/rl#1221]
## Describe the bug

The PPO example appears to make no training progress after running to completion.  As shown in the graphs below, the average reward never really increases, which I assume it is supposed to do.  The experiment hyperparameters were left as default. I seem to have the same issue with other environments with discrete actions and pixel observations such as `CartPole-v1`, but not when `from_pixels` is false or a continuous action space is used.

## To Reproduce

Run `examples\ppo\ppo.py` with the following `config.yaml`:

```
env:
  env_name: PongNoFrameskip-v4
  env_task: ''
  env_library: gym
  frame_skip: 4
  num_envs: 8
  noop: 1
  reward_scaling: 1.0
  from_pixels: true
  n_samples_stats: 3
collector:
  frames_per_batch: 4096
  total_frames: 40000000
  collector_device: cpu
  max_frames_per_traj: -1
logger:
  backend: tensorboard
  exp_name: ppo_pong_gym
  log_interval: 10000
optim:
  device: cpu
  lr: 0.00025
  weight_decay: 0.0
  lr_scheduler: true
loss:
  gamma: 0.99
  mini_batch_size: 1024
  ppo_epochs: 3
  gae_lamdda: 0.95
  clip_epsilon: 0.1
  critic_coef: 0.5
  entropy_coef: 0.01
  loss_critic_type: l2
```

## Screenshots

![graphs](https://github.com/pytorch/rl/assets/24553696/907a92ad-5ec7-4d2a-ba51-1743832c93dc)

## System info

TorchRL library cloned from GitHub, main branch.
Python version 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)] win32
Windows 10, CUDA 11.8

Pip packages:
```
ale-py                 0.8.1
antlr4-python3-runtime 4.9.3
appdirs                1.4.4
certifi                2022.12.7
charset-normalizer     2.1.1
click                  8.1.3
cloudpickle            2.2.1
colorama               0.4.6
docker-pycreds         0.4.0
filelock               3.9.0
gitdb                  4.0.10
GitPython              3.1.31
gym                    0.26.2
gym-notices            0.0.8
hydra-core             1.3.2
idna                   3.4
importlib-metadata     6.6.0
importlib-resources    5.12.0
Jinja2                 3.1.2
MarkupSafe             2.1.2
mpmath                 1.2.1
networkx               3.0
numpy                  1.24.1
omegaconf              2.3.0
packaging              23.1
pathtools              0.1.2
Pillow                 9.3.0
pip                    23.1.2
protobuf               4.23.1
psutil                 5.9.5
PyYAML                 6.0
requests               2.28.1
sentry-sdk             1.23.1
setproctitle           1.3.2
setuptools             49.2.1
six                    1.16.0
smmap                  5.0.0
sympy                  1.11.1
tensordict             0.1.2
tensordict-nightly     2023.5.15
torch                  2.0.1+cu118
torchaudio             2.0.2+cu118
torchrl                0.1.1+3fb2d0d
torchvision            0.15.2
tqdm                   4.65.0
typing_extensions      4.4.0
urllib3                1.26.13
viskit                 0.1
wandb                  0.15.3
zipp                   3.15.0
```

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

cc @albertbou92

**Comment by vmoens** [comment]
Thanks for reporting
A couple of weeks back we had good perf so I guess it must be a decent change we made.
I'll sort this out asap!

**Comment by vmoens** [comment]
This should be fixed by #1225 but let's wait until it's polished
Here's my 1M perf on Pong
![image](https://github.com/pytorch/rl/assets/25529882/e8fe19a4-a5af-4229-8e38-cc6074e017a0)

**Comment by SiliconSloth** [comment]
It seems to work now, thanks for fixing it so fast!
